spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AbstractAmqpOutboundEndpoint.java

```diff
@@ -193,10 +194,22 @@ public void setLazyConnect(boolean lazyConnect) {
 	}
 
 	/**
+	 * Set the value to set in the {@code x-delay} header when using the
+	 * RabbitMQ delayed message exchange plugin. By default, the {@link AmqpHeaders#DELAY}
+	 * header (if present) is mapped; setting the delay here overrides that value.
+	 * @param delayExpression the expression.
+	 * @since 5.0
+	 */
+	public void setDelay(int delay) {
+		this.delayExpression = new SupplierExpression<>(() -> delay);
```

Comment:
Good idea, of course, and it really has permissions to live :smile: 
But I think an existing `ValueExpression<V>` would be better here.

And since the setter is here already, WDYT about one more `delay` XSD attribute ?

In addition need to consider to move all this "strange" `Expression` implementations to SF SpEL module.